test/plots/barPlotTests.ts

```diff
@@ -1 +1,783 @@
+          }
+          dataset = new Plottable.Dataset(data);
+          barPlot.addDataset(dataset);
+          barPlot.renderTo(svg);
+        });
+
+        function getCenterOfText(textNode: SVGElement) {
+          const plotBoundingClientRect = (<SVGElement> barPlot.background().node()).getBoundingClientRect();
+          const labelBoundingClientRect = textNode.getBoundingClientRect();
+
+          return {
+            x: (labelBoundingClientRect.left + labelBoundingClientRect.right) / 2 - plotBoundingClientRect.left,
+            y: (labelBoundingClientRect.top + labelBoundingClientRect.bottom) / 2 - plotBoundingClientRect.top
+          };
+        };
+
+        it("does not show labels by default", () => {
+          const texts = barPlot.content().selectAll("text");
+          assert.strictEqual(texts.size(), 0, "by default, no texts are drawn");
+        });
+
+        it("draws one label per datum", () => {
+          barPlot.labelsEnabled(true);
+          barPlot.labelsEnabled(true);
+          const texts = barPlot.content().selectAll("text");
+          assert.strictEqual(texts.size(), data.length, "one label drawn per datum");
+          texts.each(function(d, i) {
+            assert.strictEqual(d3.select(this).text(), data[i].value.toString(), `by default, label text is the bar's value (index ${i})`);
+          });
+        });
+
+        it("hides the labels if bars are too thin to show them", () => {
+          svg.attr(baseSizeAttr, TestMethods.numAttr(svg, baseSizeAttr) / 10);
+          barPlot.redraw();
+          barPlot.labelsEnabled(true);
+
+          const texts = barPlot.content().selectAll("text");
+          assert.strictEqual(texts.size(), 0, "no labels drawn");
+        });
+
+        it("can apply a formatter to the labels", () => {
+          barPlot.labelsEnabled(true);
+          const formatter = (n: number) => `${n}%`;
+          barPlot.labelFormatter(formatter);
+
+          const texts = barPlot.content().selectAll("text");
+          assert.strictEqual(texts.size(), data.length, "one label drawn per datum");
+          const expectedTexts = data.map((d) => formatter(d.value));
+          texts.each(function(d, i) {
+            assert.strictEqual(d3.select(this).text(), expectedTexts[i], `formatter is applied to the displayed value (index ${i})`);
+          });
+        });
+
+        it("shows labels inside or outside the bar as appropriate", () => {
+          barPlot.labelsEnabled(true);
+
+          const labels = barPlot.content().selectAll(".on-bar-label, .off-bar-label");
+          assert.strictEqual(labels.size(), data.length, "one label drawn per datum");
+
+          const bars = barPlot.content().select(".bar-area").selectAll("rect");
+          labels.each((d, i) => {
+            const labelBoundingClientRect = <any> (<SVGElement> labels[0][i]).getBoundingClientRect();
+            const barBoundingClientRect = <any> (<SVGElement> bars[0][i]).getBoundingClientRect();
```

Comment:
The later lines with `labelBoundingClientRect[valueSizeAttr]` won't work unless I do this, since Typescript doesn't consider that `valueSizeAttr` will always be either `"width"` or `"height"` and therefore equivalent to `labelBoundingClientRect.width`/`labelBoundingClientRect.height`.